File: DicewareLib/StringExtensions.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace DiceWarePasswordGenerator
{
	public static class StringExtensions
	{
		public static string SafeSubString(this string str, int start, int length)
		{
			return str.Substring(start, (int)Math.Min(length, str.Length - start));
		}
	}
}

File: DicewareLib/StringExtensions.fs
namespace DiceWarePasswordGenerator

open System

[<Runtime.CompilerServices.Extension>]
module StringExtensions =
    [<Runtime.CompilerServices.Extension>]
    let SafeSubString(str : string, start : int, length : int) =
        str.Substring(start, Math.Min(length, str.Length - start))
